src/v2/view-builder/views/consent/ConsentView.js

```diff
@@ -0,0 +1,63 @@
+import { loc } from 'okta';
+
+import BaseView from '../../internals/BaseView';
+import BaseForm from '../../internals/BaseForm';
+import ConsentViewHeader from './ConsentViewHeader';
+import ConsentViewFooter from './ConsentViewFooter';
+
+
+const Body = BaseForm.extend(
+  {
+    noButtonBar: false,
+    noCancelButton: false,
+    save: () => loc('consent.required.consentButton', 'login'),
+    cancel: () => loc('consent.required.cancelButton', 'login'),
+    title: false,
+    events: {
+      'click input[data-type="save"]': 'setConsent',
+    },
+    setConsent () {
+      this.model.set('consent', true);
+    },
+    cancelForm () {
+      // override BaseForm.prototype.cancelForm which cancels auth flow
+      this.model.set('consent', false);
+      this.options.appState.trigger('saveForm', this.model);
+    },
+  },
+);
+
+export default BaseView.extend({
+  Header: ConsentViewHeader,
+  Body,
+  initialize () {
+    BaseView.prototype.initialize.apply(this, arguments);
+
+    if (this.options.appState.get('currentFormName') === 'consent') {
```

Comment:
- `=== RemediationForm.CONSENT_ENDUSER` ?
- I'd rather to have class hierarchy as we already be able to dispatch to different class at View Factory.

```
AdminConsentView.js {
  ...
}
EnduserConsentView extends AdminContentView {
  Footer: EnduserConsentViewFooter,
}
```

Reply:
I understand and have added this change https://github.com/okta/okta-signin-widget/pull/1637/commits/d6e4a3e83458efa8e13a89b0b846cad2f755d51f